docs/types/value-types.rst

```diff
@@ -210,6 +210,11 @@ an exception to this rule.
     declare its type as ``address payable`` to make this requirement visible. Also,
     try to make this distinction or conversion as early as possible.
 
+    The distinction between ``address`` and ``address payable`` was introduced with version 0.5.0.
+    Also starting from that version, contracts do not derive from the address type, but can still be explicitly converted to
```

Comment:
I see that this was here before your PR but it seems weird. Was this actually inheritance? I always thought that it was an explicit conversion plus the presence of contract members (which has a similar effect but it's technically not the same thing):

```suggestion
    Also starting from that version, contracts are not implicitly convertible to the ``address`` type, but can still be explicitly converted to
```